### PR TITLE
feat: integrate cytoscape for graphing

### PR DESCRIPTION
**Pull request by rbasu101**
Previously, plotly was utilized in generate a visual graph for the graph app, however it was very limited in functionality and user-interactivity. This PR replaces plotly with python dash's cytoscape to generate a visual graph that has much more interactivity and future "extendability" than before.

### CODE REVIEW

File: src/dgipy/graph_app.py
Comment (mcannon068nw) [inline src/dgipy/graph_app.py:1]: how much of this file is actually changed? I am wondering if this markup is being caused by end of line characters again @jsstevenson @rbasu101

File: src/dgipy/network_graph.py
Comment (mcannon068nw) [inline src/dgipy/network_graph.py:6]: this is probably a trivial detail, but is this a value that we would want to allow the user to set somehow (just in case)? Like the default can be this but they could potentially set an environmental variable or something if they really wanted to mess with it.

Comment (mcannon068nw) [reply src/dgipy/network_graph.py:6]: I'm assuming this just helps determine the initial placement and orientation of the nodes

File: src/dgipy/network_graph.py
Comment (mcannon068nw) [inline src/dgipy/network_graph.py:112]: why the switch from private to public for `initialize_network`? (for my own understanding)

File: src/dgipy/network_graph.py
Comment (mcannon068nw) [inline src/dgipy/network_graph.py:132]: this is way more compact that before. I am pro this change to cytoscape.

How much of a dependency burden is there with dash_cytoscape? Iirc we will be modularizing this separately from the dgipy search functionality anyway, but I just want to get a sense of how much the user will have to install beforehand to use this

**Review by mcannon068nw — APPROVED** [review]
approved pending any additional feedback from @jsstevenson and your responses to comments throughout. Additionally, before merging please confirm that the `graph_app.py` differences are not an end-of-line character issue again.

**Review by jsstevenson — APPROVED** [review]
the cytoscape integration is great. We've talked about using this library on the main DGIdb frontend.

* make sure to fix the line endings issue
* it'd be nice if the dropdown menu was sorted somehow, maybe by # edges, maybe alphabetical
* it'd be nice if the central node was already selected on the graph once you select it from the dropdown menu

File: src/dgipy/graph_app.py
Comment (TheCoder-101) [reply src/dgipy/graph_app.py:1]: This should *hopefully* be the last time this problem occurs. Although I did work on this branch on the mac, I believe I merged this branch from another in-progress branch that likely had commits done to it from a windows machine. That being said, if this problem continues after this PR, then there might be a problem on my end that I have to resolve.

Comment (rbasu101) [reply src/dgipy/graph_app.py:1]: This should *hopefully* be the last time this problem occurs. Although I did work on this branch on the mac, I believe I merged this branch from another in-progress branch that likely had commits done to it from a windows machine. That being said, if this problem continues after this PR, then there might be a problem on my end that I have to resolve.

File: src/dgipy/network_graph.py
Comment (rbasu101) [reply src/dgipy/network_graph.py:6]: Changing it doesn't really make any significant changes to the graph. The seed pretty much just randomizes the node positions and orientation of the graph. The only reason it has a specified value (in this case 7) is to ensure that querying the same drugs/genes twice will output identical graphs.

We certainly could allow users to change it, but it wouldn't really help them mess with the layout of the graph in any meaningful way.

File: src/dgipy/network_graph.py
Comment (rbasu101) [reply src/dgipy/network_graph.py:132]: dash_cytoscape doesn't have much of a dependency burden. Besides dash and networkx (to provide the layout of the graph), cytoscape doesn't really have any other dependencies that we'd need a user to install for cytoscape to work.

File: src/dgipy/network_graph.py
Comment (rbasu101) [reply src/dgipy/network_graph.py:112]: Privatizing `initalize_network` in network_graph.py and referencing it in graph_app.py throws the ruff error:
 [](https://docs.astral.sh/ruff/rules/private-member-access/)

**Review by mcannon068nw — APPROVED** [review]

**Pull request merged**
